tree.rb

```diff
@@ -23,7 +21,7 @@ def create_sprout(branch, letter)
     sprout
   end
 
-  def find_sprout(branch, letter)
+  def find_or_create_sprout(branch, letter)
     branch.children.each { |node| return node if node.char == letter }
```

Comment:
And what if you will need to just find sprout, but not to create it?
I suggest you to create find_sprout method also and just to use it here